apps/crawler/src/processors/extrinsic-processor.ts

```diff
@@ -0,0 +1,170 @@
+  private processor: ScanProcessor;
+
+  constructor(
+    @InjectRepository(Extrinsic)
+    private modelRepository: Repository<Extrinsic>,
+    private utils: UtilsService,
+    protected connection: Connection,
+    protected sdkService: SdkService,
+    private processorConfigService: ProcessorConfigService,
+  ) {
+    this.processor = new ScanProcessor(
+      this.name,
+      this.connection,
+      processorConfigService.getDataSource(),
+      processorConfigService.getRange(),
+      processorConfigService.getTypesBundle(),
+    );
+
+    this.logger = new Logger('ExtrinsicProcessor');
+
+    Object.values(ExtrinsicNames).forEach((name) => {
+      this.processor.addExtrinsicHandler(name, async (ctx) => {
+        await this.upsertHandler(ctx, name);
+      });
+    });
+
+    this.logger.log('Starting processor...');
+  }
+
+  private async upsertHandler(
+    ctx: ExtrinsicHandlerContext,
+    eventName: string,
+  ): Promise<void> {
+    const { height, timestamp } = ctx.block;
+
+    const log = {
+      eventName,
+      blockNumber: height,
+      blockTimestamp: timestamp,
+    };
+
+    try {
+      const data = await this.getData(ctx);
+
+      if (data) {
+        await this.modelRepository.upsert(data, [
+          'block_number',
+          'extrinsic_index',
+        ]);
+      }
+
+      this.logger.verbose({ ...log });
+    } catch (err) {
+      this.logger.error({ ...log, error: err.message });
+    }
+  }
+
+  private async getData(ctx: ExtrinsicHandlerContext) {
+    const { block, extrinsic } = ctx;
+    const events = await this.getExtrinsicEvents(ctx);
+
+    let to_owner: null | string = null;
+    if (transferMethods.includes(extrinsic.method)) {
+      const args = extrinsic.args;
+      const ownerData = args[0] as IExtrinsicOwner;
+      if (ownerData) {
+        to_owner =
+          ownerData.value.id ||
+          ownerData.value.substrate ||
+          ownerData.value.ethereum ||
+          null;
+      }
+    }
+
+    return {
+      block_number: block.height,
+      extrinsic_index: extrinsic.indexInBlock,
+      is_signed: true,
+      signer: extrinsic.signer,
+      section: extrinsic.section,
+      method: extrinsic.method,
+      args: JSON.stringify(extrinsic.args),
+      hash: extrinsic.hash,
+      doc: JSON.stringify(extrinsic.args),
+      success: this.utils.getExtrinsicSuccess(events),
+      timestamp: `${Math.floor(block.timestamp / 1000)}`,
+      amount: String(this.utils.getExtrinsicAmount(events)),
+      fee: String(this.utils.getExtrinsicFee(events)),
+      block_index: `${block.height}-${extrinsic.indexInBlock}`,
+      to_owner,
+      signer_normalized: this.utils.normalizeSubstrateAddress(extrinsic.signer),
+      to_owner_normalized: this.utils.normalizeSubstrateAddress(to_owner),
+    };
+  }
+
+  private async getExtrinsicEvents(ctx: ExtrinsicHandlerContext) {
+    const { block, extrinsic } = ctx;
+    const sdkApi = await this.sdkService.getSdk();
```

Comment:
We are not supposed to get sdk api object from here. SdkService class created just for this. You should add new method something like `SdkService.getBlockEvents(hash)`.